mavros_extras/src/plugins/trajectory.cpp

```diff
@@ -172,50 +167,74 @@ class TrajectoryPlugin : public plugin::PluginBase {
 			trajectory.point_5[6] = acceleration_point_5.x();
 			trajectory.point_5[7] = acceleration_point_5.y();
 			trajectory.point_5[8] = acceleration_point_5.z();
-			double yaw_ned_point_5 = wrap_pi(-req->point_5.yaw + (M_PI / 2.0f));
-			trajectory.point_5[9] = yaw_ned_point_5;
+			trajectory.point_5[9] = wrap_pi(-req->point_5.yaw + (M_PI / 2.0f));
 			trajectory.point_5[10] = req->point_5.yaw_rate;
 
-			// [[[end]]] (checksum: c19ffbd7dee05e9242b465c7642c3656)
+			// [[[end]]] (checksum: de8ef970a43d4b07fb7581c856c95980)
 		} else {
 			//[[[cog:
-			//for p in "12345":
-			//     cog.outl("trajectory.point_{p}[3] = req->point_{p}.velocity.x;".format(**locals()))
-			//     cog.outl("double yaw_ned_point_{p} = wrap_pi(-req->point_{p}.yaw + (M_PI / 2.0f)); \n".format(**locals()))
-			//     cog.outl("trajectory.point_{p}[4] = yaw_ned_point_{p};".format(**locals()))
-			//     cog.outl("trajectory.point_{p}[5] = req->point_{p}.yaw_rate; \n".format(**locals()))
+			//for p, index in zip("12345", "01234"):
+			//     cog.outl("trajectory.point_{p}[3] = req->time_horizon[{index}];".format(**locals()))
 			// ]]]
-			trajectory.point_1[3] = req->point_1.velocity.x;
-			double yaw_ned_point_1 = wrap_pi(-req->point_1.yaw + (M_PI / 2.0f));
-
-			trajectory.point_1[4] = yaw_ned_point_1;
-			trajectory.point_1[5] = req->point_1.yaw_rate;
-
-			trajectory.point_2[3] = req->point_2.velocity.x;
-			double yaw_ned_point_2 = wrap_pi(-req->point_2.yaw + (M_PI / 2.0f));
+			trajectory.point_1[3] = req->time_horizon[0];
+			trajectory.point_2[3] = req->time_horizon[1];
+			trajectory.point_3[3] = req->time_horizon[2];
+			trajectory.point_4[3] = req->time_horizon[3];
+			trajectory.point_5[3] = req->time_horizon[4];
+			// [[[end]]] (checksum: 380f20e816c51b6aa569cade870c5ec7)
 
-			trajectory.point_2[4] = yaw_ned_point_2;
-			trajectory.point_2[5] = req->point_2.yaw_rate;
 
-			trajectory.point_3[3] = req->point_3.velocity.x;
-			double yaw_ned_point_3 = wrap_pi(-req->point_3.yaw + (M_PI / 2.0f));
-
-			trajectory.point_3[4] = yaw_ned_point_3;
-			trajectory.point_3[5] = req->point_3.yaw_rate;
+			//[[[cog:
+			//for p in "12345":
+			//     cog.outl("trajectory.point_{p}[4] = wrap_pi(-req->point_{p}.yaw + (M_PI / 2.0f));".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[5] = NAN;".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[6] = NAN;".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[7] = NAN;".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[8] = NAN;".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[9] = NAN;".format(**locals()))
+			//     cog.outl("trajectory.point_{p}[10] = NAN;\n".format(**locals()))
+			// ]]]
```

Comment:
That looks like may be better write that way:
```cpp

auto fill_points_variant = [](std::array<float, N?> &point, double yaw) {
  // cog...
}

// cog...
fill_points_variant(traectory.point_1, req->point_1.yaw);
```

Reply:
I am not sure I understand what you would like to achieve. Would you like to have a separate method to fill each of the field, e.g. yaw, yaw_speed, etc?

Reply:
With lambda function you can reduce amount of generated code. That may look more readable.